synse/http_integration_test.go

```diff
@@ -0,0 +1,551 @@
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(plugins))
+
+	for _, plugin := range plugins {
+		assert.NotEmpty(t, plugin.Name)
+		assert.NotEmpty(t, plugin.Maintainer)
+		assert.NotEmpty(t, plugin.Tag)
+		assert.NotEmpty(t, plugin.Description)
+		assert.NotEmpty(t, plugin.ID)
+		assert.True(t, plugin.Active)
+	}
+}
+
+func TestIntegration_PluginInfo(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	plugins, err := client.Plugins()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(plugins))
+
+	for _, p := range plugins {
+		plugin, err := client.Plugin(p.ID)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, plugin.Name)
+		assert.NotEmpty(t, plugin.Maintainer)
+		assert.NotEmpty(t, plugin.Tag)
+		assert.NotEmpty(t, plugin.Description)
+		assert.NotEmpty(t, plugin.VCS)
+		assert.NotEmpty(t, plugin.ID)
+		assert.True(t, plugin.Active)
+		assert.NotEmpty(t, plugin.Network.Address)
+		assert.NotEmpty(t, plugin.Network.Protocol)
+		assert.NotEmpty(t, plugin.Version.PluginVersion)
+		assert.NotEmpty(t, plugin.Version.SDKVersion)
+		assert.NotEmpty(t, plugin.Version.BuildDate)
+		assert.NotEmpty(t, plugin.Version.GitCommit)
+		assert.NotEmpty(t, plugin.Version.GitTag)
+		assert.NotEmpty(t, plugin.Version.Arch)
+		assert.NotEmpty(t, plugin.Version.OS)
+		assert.NotEmpty(t, plugin.Health.Timestamp)
+		assert.Equal(t, "OK", plugin.Health.Status)
+		assert.Equal(t, 2, len(plugin.Health.Checks))
+
+		for _, check := range plugin.Health.Checks {
+			assert.NotEmpty(t, check.Name)
+			assert.Equal(t, "OK", check.Status)
+			assert.NotEmpty(t, check.Type)
+			assert.NotEmpty(t, check.Timestamp)
+			assert.Empty(t, check.Message)
+		}
+
+	}
+}
+
+func TestIntegration_PluginHealth(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	health, err := client.PluginHealth()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, health.Status)
+	assert.NotEmpty(t, health.Updated)
+	assert.Equal(t, 1, len(health.Healthy))
+	assert.Equal(t, 0, len(health.Unhealthy))
+	assert.Equal(t, 1, health.Active)
+	assert.Equal(t, 0, health.Inactive)
+}
+
+func TestIntegration_Scan(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.ScanOptions{}
+	devices, err := client.Scan(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 22, len(devices))
+
+	for _, device := range devices {
+		assert.NotEmpty(t, device.ID)
+		assert.NotEmpty(t, device.Info)
+		assert.NotEmpty(t, device.Type)
+		assert.NotEmpty(t, device.Plugin)
+		assert.NotEmpty(t, device.Tags)
+
+		// only led devices have alias
```

Comment:
I'm not sure I'd hardcode this check into the test, as this is liable to change given a simple change to the plugin configuration. If we run the integration tests using a custom device configuration where we can guarantee that this wont change, then I think this would be fine, but since it appears we are just using the default baked-in config, I think this may lead to issues down the road when it inevitability changes.